environment_variable.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2016 Rohith All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package marathon
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// EnvironmentVariable is an environment variable associated with an application
+type EnvironmentVariable struct {
```

Comment:
Thanks for the extra info. I've had a bit of time to ponder about the design.

From a UX perspective, my impression is that a significant number of users will not want/have to process secrets (neither from reading nor writing angle). Thus, and IMHO, it'd be convenient if we could maintain our existing environment variable map holding just non-secret environment variables and have another one holding everything related to secrets data (environment variable "pointers" and the secret sources, i.e., the kind of struct you were suggesting).

IIUC, the current design requires users interesting in processing all non-secret environment variables to change from something like

```go
for name, value := range app.Env {
    // process name/value
}
```

to 

```go
for name, kind := range app.Env {
    // Make sure this isn't a secret env var.
    if kind.Secret == "" {     
        // process name/kind.EnvVar
    }
}
```

which looks a bit clumsy and non-intuitive to me.

Personally I think it's acceptable to do a bit more heavy-lifting on the library end if the result is more easily approachable data structures to work with.

WDYT?